### ISSUE TITLE
discord accounts CAN send EMBEDS

### ISSUE THREAD

**Issue by TerminalPR0** [V4NSH4J/discord-mass-DM-GO#489]
I will refer you to my API for this: http://embd.tk

**Comment by itschasa** [comment]
website down? i can't access it.

currently the only way for a user account to send an embed is via a website url. if this site just allows people to make a link for a custom embed, then people can just put the url in the message, no changes to the program needed.

**Comment by TerminalPR0** [comment]
the website is up and running again. now you can see how it works.

чт, 7 апр. 2022 г., 19:50 chasa ***@***.***>:

**Comment by itschasa** [comment]
seems like it's what i said above. as it's an api it could be implemented, but the api would have to be online 24/7 for dmdgo users to be able to use it at any time of the day.

**Comment by TerminalPR0** [comment]
it's online 24/7, don't worry

чт, 7 апр. 2022 г., 20:22 chasa ***@***.***>:

**Comment by itschasa** [comment]
also there's a element of security involved as well.

you have to store the embed data, and theres a chance you could be storing IP addresses with them. you could then use the embed data to identify users, and use the IP address for a number of things. i'm not saying you are, but there's every chance that you could be. and i'm sure vanshaj wouldn't be very happy about adding a potential ip logger to his open source software.

**Comment by TerminalPR0** [comment]
this api does not collect data about users, except for what they themselves
send to the api

пт, 8 апр. 2022 г., 00:11 chasa ***@***.***>:

**Comment by itschasa** [comment]
yes but we can only take your word for that, we can't exactly check.

**Comment by TerminalPR0** [comment]
I, as the owner of this api, declare this.

**Comment by TerminalPR0** [comment]
я добавлю политику конфиденциальности и условия использования сервиса после того, как я сделаю обновление с аккаунтами (подробнее вы можете узнать по этой ссылке: https://t.me/discord_embed/23)

**Comment by itschasa** [comment]
you're basically saying stuff without meaning anything. it doesn't matter that you say you won't, you still could, and everyone would have no way of telling if you are or not.

it's like a scammer saying "yes i promise i will deliver the goods" and then they just don't. just because they promised they would, doesn't mean they would actually deliver.

**Comment by TerminalPR0** [comment]
if you are afraid that I will get your ip - use vpn (proxy for an api request). What else can I get with an incoming request to api, besides the entered data, IP address, user agent, what do you think?

**Comment by itschasa** [comment]
of course i would use a VPN for software i don't 100% know what it's doing, but what about the other 37,000 people who have downloaded DMDGO. that's a huge amount of people who trust in the software, and would trust in future updates. they wouldn't use a VPN on trusted software. what if they aren't aware of this entire website but they still want an embed, then they could have their ip + embed data collected without even realising.

another point is hosting. you could be getting 100,000s of requests. that's a lot of requests to handle. and people aren't going to want to use it if it takes 5m to get an embed. also hosting isn't free.

**Comment by itschasa** [comment]
@V4NSH4J what do you think of all this?

**Comment by TerminalPR0** [comment]
> of course i would use a VPN for software i don't 100% know what it's doing, but what about the other 37,000 people who have downloaded DMDGO. that's a huge amount of people who trust in the software, and would trust in future updates. they wouldn't use a VPN on trusted software. what if they aren't aware of this entire website but they still want an embed, then they could have their ip + embed data collected without even realising.
> 
> another point is hosting. you could be getting 100,000s of requests. that's a lot of requests to handle. and people aren't going to want to use it if it takes 5m to get an embed. also hosting isn't free.

my hosting is quite powerful. I think he will be up to the task.

**Comment by TerminalPR0** [comment]
if my hosting stops coping, I will fix it by increasing the hosting resources.

**Comment by itschasa** [comment]
well from what i've gathered, i think that this probably won't be added, because of 2 main factors:
- **control**, v4n would have no control over what happens with the embeds.
- **security**, giving 37k people access to embeds wouldn't be the best idea, considering no one can see what's going on behind the scenes. considering this is also an open source project, it would only make sense for everything included with this project to also be open source.

**Comment by itschasa** [comment]
> my hosting is quite powerful. I think he will be up to the task.

also considering you are using a .tk domain, which is a free domain, i have reason to doubt this.

**Comment by TerminalPR0** [comment]
I can easily prevent this (many requests). discord caches website embeds, and if you send the same link many times, discord will not contact my site for data, it will take from the cache. and in order not to clog the api, you can save the link to a variable and send this link from the variable, rather than creating a new one through the api. embeds are saved forever and moderated. they can be removed, the IP address may be restricted in access to the site for a while if it creates a very large load on the site. I can list for a very long time, but it does not make sense.

**Comment by TerminalPR0** [comment]
> well from what i've gathered, i think that this probably won't be added, because of 2 main factors:
> 
> * **control**, v4n would have no control over what happens with the embeds.
> * **security**, giving 37k people access to embeds wouldn't be the best idea, considering no one can see what's going on behind the scenes. considering this is also an open source project, it would only make sense for everything included with this project to also be open source.

about control: if he wants, I can give him such an opportunity.
about security: I will share with the creator of this project the source code of the site, if he wants.

**Comment by TerminalPR0** [comment]
> > my hosting is quite powerful. I think he will be up to the task.
> 
> also considering you are using a .tk domain, which is a free domain, i have reason to doubt this.

can't afford paid domain. almost all money goes to hosting.

**Comment by itschasa** [comment]
> I can easily prevent this (many requests). discord caches website embeds, and if you send the same link many times, discord will not contact my site for data, it will take from the cache. and in order not to clog the api, you can save the link to a variable and send this link from the variable, rather than creating a new one through the api. embeds are saved forever and moderated. they can be removed, the IP address may be restricted in access to the site for a while if it creates a very large load on the site. I can list for a very long time, but it does not make sense.

i know discord caches embed data, but more from the amount of people making the embeds. saving the link as a variable wouldn't really work, say you wanted to update the discord link on it, new embed will be needed.

**Comment by itschasa** [comment]
> > > my hosting is quite powerful. I think he will be up to the task.
> 
> > 
> 
> > also considering you are using a .tk domain, which is a free domain, i have reason to doubt this.
> 
> 
> 
> can't afford paid domain. almost all money goes to hosting.

if you can't afford a permanent domain, then the embeds will only really last until the domain expires.

**Comment by TerminalPR0** [comment]
that is, we create an embed and keep a link to it outside the loop.

**Comment by itschasa** [comment]
> it's online 24/7, don't worry
> 
> чт, 7 апр. 2022 г., 20:22 chasa ***@***.***>:

also it's offline again lol

**Comment by TerminalPR0** [comment]
hosting expired.

сб, 9 апр. 2022 г., 18:19 chasa ***@***.***>:

**Comment by V4NSH4J** [comment]
The idea is good, the implementation is kinda dodgy, it has been offline most of the times I've checked. I don't know why someone would want to force embeds as they'd appear as a link in the message text but most people still seem to want it.

**Comment by TerminalPR0** [comment]
the api is offline since I had to redo the updates that were made earlier.
since the hosting deleted all site files for non-payment, all site updates
were lost. I am working on this situation, looking for a new hosting,
restoring updates. You can follow the progress of work in our telegram
channel: @discord_embed (I'm not advertising, I just let you know where you
can track everything that happens with this project)

вс, 17 апр. 2022 г., 23:39 Vanshaj ***@***.***>:

**Comment by TerminalPR0** [comment]
by the way, I finished the job, and the api is completely online: https://discord-embed-api-site.herokuapp.com/
IMPORTANT NOTE: PLS READ https://t.me/discord_embed Telegram channel FOR FULLY UNDERSTANDING OF THE SITUATION.
